### ISSUE TITLE
How to convert keras model with custom layer to onnx model?

### ISSUE THREAD

**Issue by alexcruz0202** [onnx/keras-onnx#517]
Hello, 
Is is possible to convert follow keras model to onnx model with this tool?
The model has customized layer with "disparity" layer named.
https://github.com/yasar-rehman/SLNET/blob/master/SLNET_A_GAP_app_disp.py

While I have model conversion, met the errors.
Any clue may be help.

Thanks for developer teams.

**Comment by jiafatom** [comment]
Use `set_converter` function, see mask_rcnn converter for example, [here](https://github.com/onnx/keras-onnx/blob/master/applications/mask_rcnn/mask_rcnn.py#L673-L674)

**Comment by alexcruz0202** [comment]
Hi, @jiafatom . Thanks for your reply.
I'll try it.

**Comment by liob** [comment]
@jiafatom does this also apply for tf.keras (2.1) models? I have implemented set_converter for a custom layer (tensorflow.keras.layers.Layer), however, the converter function never gets called.

**Comment by jiafatom** [comment]
It should work. See the example above, you need call `set_converter` before `convert_keras`. Then the code will run to `get_converter` to get the custom converter for you. Try `debug_mode=True` in `convert_keras` to see if the log is helpful. Make sure  you write the correct layer type.

**Comment by liob** [comment]
@jiafatom: I did insert a print statement in keras2onnx.funcbook.get_converter to see what op_type gets requested. There is a lot of clutter, however, I can see, that the tensorflow ops used in the custom layer are looked up but not the tf.keras custom layer itself. I have implemented the tf.keras custom layer according to this [guide]. I load the saved hdf5 keras model with a custom_objects dict, providing the tf.keras custom layer.

After reading the [examples], I thought, that I could provide the custom layer class either as string (the same string which I used in the custom_objects dict) or as reference to `set_converter`. This would allow me to provide a converter implementation for the complete custom layer, not for every tf.op used in the custom layer separately.

What do I need to do in order to get the custom tf.keras layer be recognized as one unit?

[guide]: https://www.tensorflow.org/tutorials/customization/custom_layers#implementing_custom_layers
[examples]: https://github.com/onnx/keras-onnx/blob/master/applications/mask_rcnn/mask_rcnn.py#L673-L674
